src/state.rs

```diff
@@ -0,0 +1,5 @@
+use sqlx::postgres::PgPool;
+
+pub struct State {
+    pub db: PgPool,
+}
```

Comment:
Why doesn't `State` have a "constructor"?

`State` should be responsable of initializing itself and validate their dependencies.

Reply:
Sure!